### PR TITLE
PR: Don't show a file in the Editor immediately after it's selected in the switcher.

### PR DESCRIPTION
**Pull request by juanis2112**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)
* [ ] Included a screenshot (if affecting the UI)

I implemented an enter function so that users have to press enter to switch files in the file switcher and it does not displays while charging the file with the arrows.

### Issue(s) Resolved

Fixes #
Item 8 of #8580 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:
Juanis2112

### CODE REVIEW

**Comment by ccordoba12** [comment]
@juanis2112, does this PR preserve the current behavior for the Symbol switcher? I mean, the idea is that after users select a symbol, we show it immediately in the Editor, but not for files.

**Comment by juanis2112** [comment]
> @juanis2112, does this PR preserve the current behavior for the Symbol switcher? I mean, the idea is that after users select a symbol, we show it immediately in the Editor, but not for files.

@ccordoba12 Yes, it preserves the current behavior. It still shows the symbol immediately after selecting it.

File: spyder/widgets/fileswitcher.py [hunk 1]
```diff
@@ -609,13 +612,8 @@ def item_selection_changed(self):
                 try:
                     stack_index = self.paths.index(self.filtered_path[row])
                     self.plugin = self.widgets[stack_index][1]
-                    plugin_index = self.plugins_instances.index(self.plugin)
                     # Count the real index in the tabWidget of the
                     # current plugin
```
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: @juanis2112 please move these comments to the `enter` method that you created

**Comment by ccordoba12** [comment]
> @ccordoba12 Yes, it preserves the current behavior.

Great, thanks a lot @juanis2112!

File: spyder/widgets/fileswitcher.py [hunk 2]
```diff
@@ -190,20 +191,20 @@ def recurse_level(level_idx):
 class KeyPressFilter(QObject):
     """Use with `installEventFilter` to get up/down arrow key press signal."""
     UP, DOWN = [-1, 1]  # Step constants
-
     sig_up_key_pressed = Signal()
     sig_down_key_pressed = Signal()
     sig_enter_key_pressed = Signal()
+    sig_mouse_clicked = Signal()
+    itemClicked = Signal()
```
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: @juanis2112 please remove this if not needed

File: spyder/widgets/fileswitcher.py [hunk 3]
```diff
@@ -226,6 +227,7 @@ def focusOutEvent(self, event):
 class FileSwitcher(QDialog):
     """A Sublime-like file switcher."""
     sig_goto_file = Signal(int, object)
+    sig_goto_file2 = Signal(object)
```
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: @juanis2112 please remove this if is not needed

File: spyder/widgets/fileswitcher.py [hunk 4]
```diff
@@ -190,20 +191,20 @@ def recurse_level(level_idx):
 class KeyPressFilter(QObject):
     """Use with `installEventFilter` to get up/down arrow key press signal."""
     UP, DOWN = [-1, 1]  # Step constants
-
```
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: @juanis2112 please undo the deletion of the blank line

File: spyder/widgets/fileswitcher.py [hunk 5]
```diff
@@ -190,20 +191,20 @@ def recurse_level(level_idx):
...
+    itemClicked = Signal()
 
     def eventFilter(self, src, e):
         if e.type() == QEvent.KeyPress:
             if e.key() == Qt.Key_Up:
                 self.sig_up_key_pressed.emit()
             elif e.key() == Qt.Key_Down:
                 self.sig_down_key_pressed.emit()
-            elif (e.key() == Qt.Key_Enter) or (e.key() == Qt.Key_Return):
+            elif (e.key() == Qt.Key_Return):
```
Comment (dalthviz) [inline spyder/widgets/fileswitcher.py]: @juanis2112 did you tested this on Mac, Linux and Windows?

Comment (juanis2112) [reply spyder/widgets/fileswitcher.py]: @dalthviz yes, it works for the three of them.

**Pull request merged**